Factorial/src/Factorial.java

```diff
@@ -0,0 +1,80 @@
+import org.junit.jupiter.api.Test;
+import org.junit.runner.JUnitCore;
+import org.junit.runner.Result;
+import org.junit.runner.notification.Failure;
+import static org.junit.jupiter.api.Assertions.*;
+
+/* Reads an integer and prints n! = 1 * 2 * ... * n to standard output
+   For eg: 0! = 1, 1! = 1, 2! = 2 * 1 = 2, 3! = 3 * 2 * 1 = 6 ..  so on
+   Remarks
+    - There would be an overflow for long if n > 20
+    (Factorial for 21 will give you -> -4249290049419214848 which is incorrect)
+*/
+public class Factorial {
+    // Return n!
+    public static long factorialUsingRecursion(long n) {
+        if (n < 0) {
+            throw new RuntimeException("Underflow error in factorial");
+        }
+        else if (n > 20) {
+            throw new RuntimeException("Overflow error in factorial");
+        }
+        else if (n == 0) {
+            return 1;
+        } else {
+            return n * factorialUsingRecursion(n - 1);
+        }
+    }
+
+    @Test
+    public void factorialForAnyNumTest() {
+        long num = 4;
+        long actual = factorialUsingRecursion(num);
+        long expected = 24;
+        assertEquals(actual, expected);
```

Comment:
Too many unnecessary variables. Just do: 
```java
assertEquals(factorialUsingRecursion(4), 24);
```

Reply:
Isn't it good to have actual and expected variables while writing tests?
That is a nice to have practice usually? Correct me if I am wrong.
I would remove the num variable, yes that can be passed to a method directly